crates/index-scheduler/src/scheduler/process_batch.rs

```diff
@@ -314,6 +315,26 @@ impl IndexScheduler {
                 task.status = Status::Succeeded;
                 Ok(vec![task])
             }
+            Batch::UpgradeDatabase { mut tasks } => {
+                let ret = catch_unwind(AssertUnwindSafe(|| self.process_upgrade(progress)));
+                match ret {
+                    Ok(Ok(())) => (),
+                    Ok(Err(e)) => return Err(Error::TaskDatabaseUpgrade(Box::new(e))),
+                    Err(_e) => {
+                        return Err(Error::TaskDatabaseUpgrade(Box::new(
+                            Error::ProcessBatchPanicked,
```

Comment:
todo in a new issue: have `ProcessBatchPanicked` take the payload, and rebuild a message from it (look in `crates/xtask/src/bench` for way to downcast a panic payload)

Reply:
I opened the issue for the v1.13